src/GraphQL/Execution/DocumentExecuter.cs

```diff
@@ -74,11 +89,23 @@ public virtual async Task<ExecutionResult> ExecuteAsync(ExecutionOptions options
                 }
 
                 var document = options.Document;
+                bool saveInCache = false;
+                bool analyzeComplexity = true;
+                var validationRules = options.ValidationRules;
                 using (metrics.Subject("document", "Building document"))
```

Comment:
The line `  using (metrics.Subject("document", "Analyzing complexity"))` now is under `if analyzeComplexity` condition. Should it be done for `using (metrics.Subject("document", "Building document"))` ?

Reply:
I was thinking about that also. Either way is fine to me. Since the old behavior when supplying options.Document still logged the metric, I left it that way. There certainly is a very little time required for the dictionary lookup so maybe it should stay this way. What do you suggest?

Reply:
OK. Let's keep it as is since dictionary lookup is in fact some arbitrary code and in case of distributed dictionary (Redis, etc.) it _may_ run indefinitely long.